src/GlobalContext/index.js

```diff
@@ -0,0 +1,679 @@
+import { createContext, useState, useEffect, useContext } from "react";
+import Cookies from "js-cookie";
+import { StateManager } from "../App";
+import { useNavigate } from "react-router-dom";
+
+export const GlobalProvider = createContext();
+export const GlobalContext = (props) => {
+  const navigate = useNavigate();
+  const { setToken } = useContext(StateManager);
+  const [theme, setTheme] = useState("light");
+  useEffect(() => {
+    if (theme === "dark") {
+      document.documentElement.classList.add("dark");
+    } else {
+      document.documentElement.classList.remove("dark");
+    }
+  }, [theme]);
+  const handleTheme = () => {
+    setTheme(theme === "dark" ? "light" : "dark");
+  };
+  const [userData, setUserData] = useState({
+    user: Cookies.get("user") ? JSON.parse(Cookies.get("user")) : null,
+  });
+  const [userToken, setUserToken] = useState(Cookies.get("token"));
+  const [isSidebarFull, setIsSidebarFull] = useState(false);
+  const [isEditingTask, setIsEditingTask] = useState(false);
+  const [isEditingProject, setIsEditingProject] = useState(false);
+  const handleLogout = () => {
+    Cookies.remove("user");
+    Cookies.remove("token");
+  };
+  const [currentYear, setCurrentYear] = useState(new Date().getFullYear());
+  const [currentMonth, setCurrentMonth] = useState(new Date().getMonth());
+  const [months, setMonth] = useState([
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ]);
+  const [days, setDays] = useState([
+    "Sun",
+    "Mon",
+    "Tue",
+    "Wed",
+    "Thu",
+    "Fri",
+    "Sat",
+  ]);
```

Comment:
why do you use a useState for these?
I suggest that you just use a constant instead of useState because months and day names are never gonna change. But you ain't gonna need this if you installed `dayjs`

Reply:
mantapp mas wicak ripiu nya wkwk

Reply:
Thanks a lot for the feedbacks kak